__tests__/integration.test.js

```diff
@@ -149,6 +150,61 @@ describe("/api/articles/:article_id/comments", () => {
                 expect(response.body.msg).toBe("Invalid Request");
             });
     });
+    test("POST:201 Should insert the new comment into the database and send the object back to the client", () => {
+        return request(app)
+            .post("/api/articles/2/comments")
+            .send({
+                username: "rogersop",
+                body: "This is a test comment...",
+            })
+            .expect(201)
+            .then((response) => {
+                const comment = response.body.comment;
+                expect(Object.keys(comment).length).toBe(6);
+                expect(typeof comment.comment_id).toBe("number");
+                expect(typeof comment.votes).toBe("number");
+                expect(typeof comment.created_at).toBe("string");
+                expect(typeof comment.author).toBe("string");
+                expect(typeof comment.body).toBe("string");
+                expect(typeof comment.article_id).toBe("number");
+            });
+    });
+    test("POST:404 If a valid id is given, but it does not exist in the database, an error message should be sent", () => {
+        return request(app)
+            .post("/api/articles/9999/comments")
+            .send({
+                username: "rogersop",
+                body: "This is a test comment...",
+            })
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe("Article not found");
+            });
+    });
+    test("POST:400 If an invalid id is given, an arror message should be sent", () => {
+        return request(app)
+            .post("/api/articles/dodgyID/comments")
+            .send({
+                username: "rogersop",
+                body: "This is a test comment...",
+            })
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe("Invalid Request");
+            });
+    });
+    test("POST:400 If an invalid request body is given, an error message should be sent", () => {
```

Comment:
What about if we try to post a comment as a user that doesn't exist? Hint: write your test and check for errors in your catch/middleware funcs 🧠